hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/SwapBlockListOp.java

```diff
@@ -130,15 +129,10 @@ private static void validateInode(INodesInPath srcIIP)
 
     String errorPrefix = "DIR* FSDirectory.swapBlockList: ";
     String error = "Swap Block List input ";
-    final INode srcInode = srcIIP.getLastINode();
 
-    // Check if INode is null.
-    if (srcInode == null) {
-      error  += srcIIP.getPath() + " is not found.";
-      NameNode.stateChangeLog.warn(errorPrefix + error);
-      throw new FileNotFoundException(error);
-    }
+    FSDirectory.resolveLastINode(srcIIP);
 
+    final INode srcInode = srcIIP.getLastINode();
```

Comment:
Any reason for not assigning FSDirectory.resolveLastINode(srcIIP); to srcInode?

Reply:
Thanks, good catch. Missed it.